From: Gavin Lambert <gavinl@compacsort.com>
Date: Tue, 25 Jul 2017 13:23:52 +1200

Use size_t rather than ssize_t for ec_mac_print.

Since global.h is included from usermode code, which doesn't know about ssize_t
without an additional header, and because there's no reason for this code to be
using ssize_t or off_t in the first place, let's just use size_t throughout.

diff --git a/master/globals.h b/master/globals.h
--- a/master/globals.h
+++ b/master/globals.h
@@ -278,7 +278,7 @@ unsigned int ec_master_count(void);
 void ec_print_data(const uint8_t *, size_t);
 void ec_print_data_diff(const uint8_t *, const uint8_t *, size_t);
 size_t ec_state_string(uint8_t, char *, uint8_t);
-ssize_t ec_mac_print(const uint8_t *, char *);
+size_t ec_mac_print(const uint8_t *, char *);
 int ec_mac_is_zero(const uint8_t *);
 
 ec_master_t *ecrt_request_master_err(unsigned int);
diff --git a/master/module.c b/master/module.c
--- a/master/module.c
+++ b/master/module.c
@@ -239,12 +239,12 @@ int ec_mac_equal(
  *
  * \return number of bytes written.
  */
-ssize_t ec_mac_print(
+size_t ec_mac_print(
         const uint8_t *mac, /**< MAC address */
         char *buffer /**< Target buffer. */
         )
 {
-    off_t off = 0;
+    size_t off = 0;
     unsigned int i;
 
     for (i = 0; i < ETH_ALEN; i++) {
